packages/block-editor/src/hooks/style.js

```diff
@@ -312,6 +319,15 @@ export function addSaveProps(
 		}
 	} );
 
+	/*
+	 * Set styles defaults.
+	 * Applies default values to the style object based on the block settings.
+	 * Only applies to background styles for now.
+	 */
+	if ( !! style.background ) {
+		style = setBackgroundStyleDefaults( style, editorSettings );
```

Comment:
The rationale is that some style values need to have sensible defaults that are output as CSS, but not necessarily part of the styles attribute. 

`setBackgroundStyleDefaults` could one day be `setStylesDefaults` or something.

This could theoretically be done in the style engine if we wanted to make the style engine opinionated.